src/meta-srv/src/service/admin/maintenance.rs

```diff
@@ -48,79 +49,120 @@ impl TryFrom<MaintenanceResponse> for String {
-                if params.is_empty() {
-                    self.get_maintenance().await
-                } else {
+                if path.ends_with(STATUS_SUFFIX) {
+                    // Handle GET request to '/admin/maintenance/status'
+                    let response = self.get_maintenance().await?;
+                    to_json_response(response)
+                } else if path.ends_with(MAINTENANCE_PATH) && params.is_empty() {
+                    // Handle GET request to '/admin/maintenance'. (The legacy version)
+                    let response = self.get_maintenance().await?;
+                    to_json_response(response)
+                } else if path.ends_with(MAINTENANCE_PATH) {
+                    // Handle GET request to '/admin/maintenance' with URL parameters. (The legacy version)
                     warn!(
                         "Found URL parameters in '/admin/maintenance' request, it's deprecated, will be removed in the future"
                     );
                     // The old version operator will send GET request with URL parameters,
                     // so we need to support it.
-                    self.set_maintenance(params).await
+                    let response = self.handle_legacy_maintenance(params).await?;
+                    to_json_response(response)
+                } else {
+                    to_not_found_response()
                 }
             }
             http::Method::PUT => {
-                warn!("Found PUT request to '/admin/maintenance', it's deprecated, will be removed in the future");
-                self.set_maintenance(params).await
+                // Handle PUT request to '/admin/maintenance' with URL parameters. (The legacy version)
+                if path.ends_with(MAINTENANCE_PATH) {
+                    warn!("Found PUT request to '/admin/maintenance', it's deprecated, will be removed in the future");
+                    let response = self.handle_legacy_maintenance(params).await?;
+                    to_json_response(response)
+                } else {
+                    to_not_found_response()
+                }
+            }
+            http::Method::POST => {
+                // Handle POST request to '/admin/maintenance/enable'
+                if path.ends_with(ENABLE_SUFFIX) {
+                    let response = self.set_maintenance().await?;
+                    to_json_response(response)
+                } else if path.ends_with(DISABLE_SUFFIX) {
+                    // Handle POST request to '/admin/maintenance/disable'
+                    let response = self.unset_maintenance().await?;
+                    to_json_response(response)
+                } else if path.ends_with(MAINTENANCE_PATH) {
+                    // Handle POST request to '/admin/maintenance' with URL parameters. (The legacy version)
+                    warn!("Found PUT request to '/admin/maintenance', it's deprecated, will be removed in the future");
```

Comment:
better also add a todo to remind us to delete the related codes

Reply:
resolved in [532432c](https://github.com/GreptimeTeam/greptimedb/pull/6393/commits/532432cfc4e8ef35c09d44ba1c089b6262273452)